api/src/main/java/org/opennms/integration/api/v1/scv/SecureCredentialsVault.java

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2022 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ * OpenNMS(R) Licensing <license@opennms.org>
+ *      http://www.opennms.org/
+ *      http://www.opennms.com/
+ *******************************************************************************/
+package org.opennms.integration.api.v1.scv;
+
+import java.util.Set;
+
+import org.opennms.integration.api.v1.annotations.Consumable;
+
+@Consumable
+public interface SecureCredentialsVault {
+
+    Set<String> getAliases();
+    Credentials getCredentials(String alias);
```

Comment:
we don't restrict the access here. Technically the plugin can read all credentials. I am not sure if this ok. It comes down to if the admin trusts the plugins.

Reply:
we fixed that in the implementation provided by OpenNMS